pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,657 @@
+    `matplotlib docs <https://matplotlib.org/stable/tutorials/introductory/customizing.html>`_ .
+    If we customize ``plt.rcParams`` before executing our example function, we get a
+    different style:
+
+    .. code-block:: python
+
+        plt.rcParams['patch.facecolor'] = 'white'
+        plt.rcParams['patch.edgecolor'] = 'black'
+        plt.rcParams['patch.linewidth'] = 2
+        plt.rcParams['patch.force_edgecolor'] = True
+        plt.rcParams['lines.color'] = 'black'
+
+    .. figure:: ../../_static/drawer/example_rcParams.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+    Instead of manually customizing everything, you can choose one of
+    the provided styles. You can see available styles with ``plt.style.available``.
+    We can set the ``'Solarize_Light2'`` style with the same graph as drawn above and instead get:
+
+    .. code-block:: python
+
+        plt.style.use('Solarize_light2')
+
+    .. figure:: ../../_static/drawer/example_Solarize_Light2.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+    You can also manually control the styles of individual plot elements via the drawer class.
+    Any control-type method, ``ctrl``, ``CNOT``, ``_ctrl_circ``, ``_ctrlo_circ``, and ``_target_x``, only
+    accept a color keyword.  All other gates accept dictionaries of keyword-values pairs for matplotlib object
+    components.  Acceptable keywords differ based on what's being drawn. For example, you cannot pass ``"fontsize"``
+    to the dictionary controlling how to format a rectangle.
+
+    This example demonstrates the different ways you can format the individual elements:
+
+    .. code-block:: python
+
+        wire_kwargs = {"color": "indigo", "linewidth": 4}
+        drawer = MPLDrawer(n_wires=2,n_layers=4, wire_kwargs=wire_kwargs)
+
+        label_kwargs = {"fontsize": "x-large", 'color': 'indigo'}
+        drawer.label(["0","a"], text_kwargs=label_kwargs)
+
+        box_kwargs = {'facecolor':'lightcoral', 'edgecolor': 'maroon', 'linewidth': 5}
+        text_kwargs = {'fontsize': 'xx-large', 'color':'maroon'}
+        drawer.box_gate(layer=0, wires=0, text="Z", box_kwargs=box_kwargs, text_kwargs=text_kwargs)
+
+        swap_kwargs = {'linewidth': 4, 'color': 'darkgreen'}
+        drawer.SWAP(layer=1, wires=(0,1), kwargs=swap_kwargs)
+
+        drawer.CNOT(layer=2, wires=(0,1), color='teal')
+
+        drawer.ctrl(layer=3, wires=(0,1), color='black')
+
+
+        measure_box = {'facecolor': 'white', 'edgecolor': 'indigo'}
+        measure_lines = {'edgecolor': 'indigo', 'facecolor': 'plum', 'linewidth': 2}
+        for wire in range(2):
+            drawer.measure(layer=4, wires=wire, box_kwargs=measure_box, lines_kwargs=measure_lines)
+
+        drawer.fig.suptitle('My Circuit', fontsize='xx-large')
+
+    .. figure:: ../../_static/drawer/example_formatted.png
+            :align: center
```

Comment:
This figure too. I like how comprehensive this example is, this is really helpful :100: